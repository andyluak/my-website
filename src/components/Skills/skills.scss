section.skills {
	background: #2d2929;
	margin-top: -3em;
	display: flex;
	flex-direction: column;
	align-items: center;
	text-align: center;
	padding-bottom: 3em;

	p,
	h1,
	h4 {
		color: white;
	}

	h1 {
		padding: 3.5em 0 1.5em 0;
	}
	.skills-container {
		display:flex;
		flex-direction: column;
		align-items: center;
		.skill_container {
			width: 50%;
			background: #212020;
			padding: 3em;
			display: flex;
			justify-content: center;
			margin-bottom: 1.5em;

			.skill_icon {
				width: 80%;
				margin-bottom: 1em;
			}
		}
	}

}

@media screen and (min-width: 728px) {
	section.skills {
		h1 {
			font-size: 2em;
			padding: 3em 0 1.5em 0;
		}
		.skills-container {
			flex-direction: row;
			gap: 3em;
			padding: 0 3em;

			.skill_container {
				.skill_icon {
					width: 100%;
				}
				p {
					font-size: 0.75em;
				}
			}
		}
	}
}

@media screen and (min-width: 1100px) {
	section.skills {
		h1 {
			font-size: 3em;
			padding: 2em 0 1.5em 0;
		}
		.skills-container {
			flex-direction: row;
			gap: 3em;
			padding: 0 8em;

			.skill_container {
				p {
					font-size: 1em;
				}
			}
		}
	}
}
