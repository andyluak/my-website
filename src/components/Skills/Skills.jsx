import './skills.scss';
import { ReactComponent as WebDev } from '../../assets/images/webdev.svg';
import { ReactComponent as PluginDev } from '../../assets/images/plugindev.svg';
import { ReactComponent as SiteDev } from '../../assets/images/sitedev.svg';
import React from 'react';

const Skills = () => {
	return (
		<section className="skills">
			<h1>SKILLS</h1>
			<div className="skills-container">
				<div className="skill_container">
					<div className="skill">
						<WebDev className="skill_icon" />
						<p>
							Anything that lives on the web, I have an interest
							for.
						</p>
					</div>
				</div>
				<div className="skill_container">
					<div className="skill">
						<PluginDev className="skill_icon" />
						<p>
							Anything that lives on the web, I have an interest
							for.
						</p>
					</div>
				</div>
				<div className="skill_container">
					<div className="skill">
						<SiteDev className="skill_icon" />
						<p>
							Anything that lives on the web, I have an interest
							for.
						</p>
					</div>
				</div>
			</div>
		</section>
	);
};

export default Skills;
