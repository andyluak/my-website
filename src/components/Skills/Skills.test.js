import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Skills from './Skills';

describe( 'Skills component tests', () => {
	it( 'excepct to render Skills component', () => {
		expect(shallow(<Skills />)).toMatchSnapshot();
	})
})

