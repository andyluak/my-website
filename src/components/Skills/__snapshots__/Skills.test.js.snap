// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills component tests excepct to render Skills component 1`] = `
<section
  className="skills"
>
  <h1>
    SKILLS
  </h1>
  <div
    className="skills-container"
  >
    <div
      className="skill_container"
    >
      <div
        className="skill"
      >
        <ForwardRef(SvgWebdev)
          className="skill_icon"
        />
        <p>
          Anything that lives on the web, I have an interest for.
        </p>
      </div>
    </div>
    <div
      className="skill_container"
    >
      <div
        className="skill"
      >
        <ForwardRef(SvgPlugindev)
          className="skill_icon"
        />
        <p>
          Anything that lives on the web, I have an interest for.
        </p>
      </div>
    </div>
    <div
      className="skill_container"
    >
      <div
        className="skill"
      >
        <ForwardRef(SvgSitedev)
          className="skill_icon"
        />
        <p>
          Anything that lives on the web, I have an interest for.
        </p>
      </div>
    </div>
  </div>
</section>
`;
